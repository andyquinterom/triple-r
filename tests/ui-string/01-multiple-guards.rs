use triple_r::ReusableString;

fn main() {
    let mut s = ReusableString::default();
    // First mutable borrow occurs here.
    let _guard1 = s.recycle();
    // This second mutable borrow should fail because `s` is already borrowed.
    let _guard2 = s.recycle();
}
