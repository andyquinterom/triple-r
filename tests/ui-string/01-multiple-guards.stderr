error[E0499]: cannot borrow `s` as mutable more than once at a time
 --> tests/ui-string/01-multiple-guards.rs:8:19
  |
6 |     let _guard1 = s.recycle();
  |                   - first mutable borrow occurs here
7 |     // This second mutable borrow should fail because `s` is already borrowed.
8 |     let _guard2 = s.recycle();
  |                   ^ second mutable borrow occurs here
9 | }
  | - first borrow might be used here, when `_guard1` is dropped and runs the `Drop` code for type `ReusableStringGuard`
